apps/prairielearn/src/middlewares/authzCourseOrInstance.js

```diff
@@ -348,6 +350,9 @@ export async function authzCourseOrInstance(req, res) {
     res.locals.is_administrator = res.locals.authz_data.is_administrator;
 
     res.locals.authz_data.mode = effectiveParams.req_mode;
+    res.locals.authz_data.mode_reason = req.cookies.pl_requested_mode
+      ? 'Requested'
```

Comment:
I think I need to think a little more about what values we want to use when overrides are in place.

I think our handling of `mode` above on line 352 is actually a little weird. `effectiveParams.req_mode` takes into account `req.cookies.pl_requested_mode`, but in this branch, we specifically don't _want_ to take that into account, does that sound right? Or... now that I think about this, I think we actually need this to take into account the cookie so that `instructorEffectiveUser` can display the effective mode. 🥴

Reply:
I guess I'm a little unclear about this myself. Maybe we do want to allow mode overrides to set `mode_reason = PrairieTest` so people can try out exam access themselves?

Reply:
Two current thoughts:

- I'm inclined to say that instructors should have to use "student view without access restrictions" if they want to preview an assessment and only have a PT access rule.
- I'm wondering if allowing instructors to override "mode" is even useful at all. That made a little more sense when one would just put `"mode": "Exam"` in an access rule and then exam network IP ranges would do the rest, but now that pretty much all Exam-mode stuff is happening through PT... it seems much less relevant now, if not completely irrelevant. Does access mode change anything user-visible that I'm forgetting about?

Reply:
At this point I think the only reason to be able to override "mode" is to allow someone to fully emulate the setup for a testing center session. But I very much doubt that anyone is doing this or that it's even needed. Do we ever need this capability ourselves? I don't think so.

Instructors can already emulate course sessions themselves because those will actually set mode:Exam so there's no need to emulate anything. And as you say, they can just use "student view without access restrictions" if they are only trying to view the assessment as a student in PL.

Given all this, I'm supportive of removing mode overrides entirely, which of course resolves a lot of the questions above 😄.

Reply:
Resolved by https://github.com/PrairieLearn/PrairieLearn/pull/10257